### PR TITLE
Ll confidence

### PR DESCRIPTION
**Pull request by rinafriedberg**
Added confidence intervals to local linear prediction and overall structure for CI to default prediction strategies.

### CODE REVIEW

File: r-package/grf/bindings/CustomForestBindings.cpp [hunk 1]
```diff
@@ -15,6 +15,7 @@
   along with grf. If not, see <http://www.gnu.org/licenses/>.
  #-------------------------------------------------------------------------------*/
 
+#include <map>
```
Comment (jtibshirani) [inline r-package/grf/bindings/CustomForestBindings.cpp]: I don't think we need to make any changes in this file.

File: r-package/grf/tests/testthat/test_local_linear_forest.R [hunk 2]
```diff
@@ -115,4 +115,72 @@ test_that("local linear predict returns local linear predictions even without tu
...
+    preds.rf = predict(forest, estimate.variance = TRUE)
+    preds.llf = predict(forest, linear.correction.variables = 1,
+                        ll.lambda = 10000000, estimate.variance = TRUE)
+
+    expect_true(max(abs(preds.llf$predictions - preds.rf$predictions)) < 10^-5)
+    expect_true(max(abs(preds.llf$variance.estimates - preds.rf$variance.estimates)) < 10^-5)
+})
+
+
+test_that("local linear confidence intervals match regression forest with large lambda", {
```
Comment (jtibshirani) [inline r-package/grf/tests/testthat/test_local_linear_forest.R]: Is this a duplicate of the method above?

**Comment by rinafriedberg** [comment]
@jtibshirani These changes should be addressed now!! Let me know if I missed any or if you'd like me to address any other comments. I added M_stored so the value of lambda would correspond to the actual regularization; it looked like otherwise we were just adding each lambda onto the other, because I modified M at each step of the loop.

File: core/src/prediction/LocalLinearPredictionStrategy.cpp [hunk 3]
```diff
@@ -52,24 +52,24 @@ std::vector<double> LocalLinearPredictionStrategy::predict(
...
   }
 
   Eigen::MatrixXd X (num_nonzero_weights, num_variables+1);
   Eigen::MatrixXd Y (num_nonzero_weights, 1);
   for (size_t i = 0; i < num_nonzero_weights; ++i) {
     for (size_t j = 0; j < num_variables; ++j){
       size_t current_predictor = linear_correction_variables[j];
-      X(i,j+1) = test_data->get(sampleID, current_predictor)
-              - original_data->get(indices[i],current_predictor);
+      X(i,j+1) = original_data->get(indices[i],current_predictor)
```
[the remark this answers is not in the document]
Comment (rinafriedberg) [reply core/src/prediction/LocalLinearPredictionStrategy.cpp]: Not sure why that showed up -- my branch should read

size_t current_predictor = linear_correction_variables[j];
X(i,j+1) = original_data->get(indices[i],current_predictor)
                 - test_data->get(sampleID, current_predictor);

I've double checked to make sure that's correct in the pushed files now!

[the remark this answers is not in the document]
Comment (rinafriedberg) [reply core/src/prediction/LocalLinearPredictionStrategy.cpp]: r.e. Custom forest bindings-  I thought I updated that file to match the current grf implementation exactly and erase any changes I made before; what changes are showing up?

**Comment by jtibshirani** [comment]
@swager committed some tests, that uncovered a discrepancy between the output of `tune_local_linear_forest` and `predict.local_linear_forest`. I fixed the issue on `master` in #324 and will merge it in.

**Pull request merged**